### PR TITLE
Changes Obsession resilience to require a lobotomy

### PR DESCRIPTION
**Pull request by RaveRadbury**
## About The Pull Request

Obsessed is changed to require a lobotomy rather than brain surgery. Also nerfs Mind Restoration so it cannot remove deep-rooted traumas.

## Why It's Good For The Game

Obsessed has been well-known for some time as being surprise ganked by fairly low-key airborne healing viruses. Players are stripped of their antag status with no fanfare or consequence.

With this change the cure for obsessed will require a lobotomy, which will cause additional permanent brain traumas. This provides way more mechanically-informed RP than losing antag due to a friendly sneeze.

## Changelog

:cl:
balance: Obsessed has been bumped up to a Deep-Rooted Brain Trauma
balance: Mind Restoration can't remove deep-rooted traumas, making Obsessed unremovable by virusses
/:cl:

### CODE REVIEW

**Comment by CRITAWAKETS** [comment]
if the point of this PR is to prevent viruses healing obsession then im sorry but
image

**Comment by RaveRadbury** [comment]
> if the point of this PR is to prevent viruses healing obsession then im sorry but image

It's my understanding that the virus will need to be more specialized towards trauma healing in order to remove it, which is less likely than the current tier it sits at.

**Comment by Redrover1760** [comment]
I like the idea of requiring a lobotomy to unobssess obssessions, its funny.

I can't believe upgraded mind restoration actually cures deep rooted traumas, I assumed it stopped at severe traumas. Then again, most people dont dedicate their virus around mind restoration, so its probably not a big deal.

**Comment by ATH1909** [comment]
> > if the point of this PR is to prevent viruses healing obsession then im sorry but image
> 
> It's my understanding that the virus will need to be more specialized towards trauma healing in order to remove it, which is less likely than the current tier it sits at.
> 
> image

the tiers here go basic -> surgery -> lobotomy, with basic being healed by mind restoration viruses with trauma_heal_mild and basic, surgery, and lobotomy being healed by mind restoration viruses with trauma_heal_severe. as such, upgrading the resilience tier of the obsession trauma from surgery to lobotomy won't actually make it any harder to heal with a virus.

**Review by Time-Green — APPROVED** [review]
Good change, virus deantagging people is dumb. Virus being able to heal deep-rooted is also dumb but it’s probably not something anyone bothers to optimize their virus towards

**Comment by SkeletalElite** [comment]
> Good change, virus deantagging people is dumb. Virus being able to heal deep-rooted is also dumb but it’s probably not something anyone bothers to optimize their virus towards

This PR won't actually change this at all as is as far as I understand it.

Mind restoration without 9 resistance will only heal brain traumas, which could be cured with neurine. (which means it wouldn't be able to remove the brain trauma with this PR)
Mind restoration with 9 resistance will still be able to remove the trauma, both before and after this PR is merged as is.

**Comment by Time-Green** [comment]
Hm yeah looks like trauma_heal_severe = TRUE actually heals deep rooted trauma’s instead of only severe trauma’s, so in effect this does not change anything with virus interactions

@RaveRadbury make mind restoration’s trauma_heal_severe be limited to severe trauma’s and not lobotomy, seems like an overpowered oversight

**Comment by RaveRadbury** [comment]
> Hm yeah looks like trauma_heal_severe = TRUE actually heals deep rooted trauma’s instead of only severe trauma’s, so in effect this does not change anything with virus interactions
> 
> @RaveRadbury make mind restoration’s trauma_heal_severe be limited to severe trauma’s and not lobotomy, seems like an overpowered oversight

In the original discussion I had with two maintainers about this we all agreed that having a specialized virus able to take care of obsession was fine, especially considering that mind restoration *reduces* resistance by -2 and a resistance of 9 is required to heal deep-rooted.

This is intentional. If a viro wants to make a 9 resistance mind restoration virus to get rid of an obsessed they are welcome to do it, however odds are it won't be a virus that's spreading among the crew.

**Comment by RaveRadbury** [comment]
> Hm yeah looks like trauma_heal_severe = TRUE actually heals deep rooted trauma’s instead of only severe trauma’s, so in effect this does not change anything with virus interactions
> 
> @RaveRadbury make mind restoration’s trauma_heal_severe be limited to severe trauma’s and not lobotomy, seems like an overpowered oversight

When I made this PR it was intended to have obsessed be curable with a resistance 9 mind restoration virus. Mind restoration already gives -2 resistance, so this is a more difficult and intentional decision. It is an accepted viable alternative to lobotomy, I was aware of it when I made the PR, which is why I said that lobotomy is the common solution (as more people will feel comfortable doing a lobotomy than making the specific virus that is needed to cure deep-rooted traumas).

**Comment by Time-Green** [comment]
@RaveRadbury 
You said in the OP you didn’t want a virus to randomly remove obsessed? You should change the changelog and PR description then

Also you already needed a 9+ res to heal obsession before this PR, so then I don’t understand what has changed accept the needed surgery? 

(6 to 9 res heals basic trauma’s, 9+ heals both lobotomy and severe, so bumping the trauma from severe to lobotomy doesn’t change anything for virus interaction)

**Comment by RaveRadbury** [comment]
@Time-Green

Alright, I see where the misunderstanding is here. In initial conversations I had with others we had assumed that the res 9 threshold would be a new thing for the trauma, when as has been pointed out here that is not the case. So yeah, oversight on my part.

Thank you for laying it out clearly.

**Comment by Time-Green** [comment]
Yeah okay then it makes sense to me too. When you’re ready make sure to update the PR description

**Comment by RaveRadbury** [comment]
> Yeah okay then it makes sense to me too. When you’re ready make sure to update the PR description

Done!

**Comment by Redrover1760** [comment]
> @Time-Green
> 
> Alright, I see where the misunderstanding is here. In initial conversations I had with others we had assumed that the res 9 threshold would be a new thing for the trauma, when as has been pointed out here that is not the case. So yeah, oversight on my part.
> 
> The ideal was that a highly-specialized virus (one that was involved enough that it couldn't be easily transmittable) would have to be developed in order to cure without needing a lobotomy. This would allow for more than one solution to an antag which would be a little bit of viro content.
> 
> Thank you for laying it out clearly. Thanks to @SkeletalElite as well! 👍

As an alternative, you could add a new resistance tier to mind restoration, like 12 or 13, that allows it to cure deep rooted traumas instead of resistance 9 allowing it to cure deep rooted traumas, or something.

**Comment by RaveRadbury** [comment]
> As an alternative, you could add a new resistance tier to mind restoration, like 12 or 13, that allows it to cure deep rooted traumas instead of resistance 9 allowing it to cure deep rooted traumas, or something. Mind restoration gives -3 transmission and -2 resistance, so I think if one were to dedicate the virus to 13+ resistance and 7 transmission total for contact transmission it wouldn't be possible. At best it'd be fluid transmission..
> 
> Edit: I've done the math if one were to optimize solely resistance and transmission and mind restoration you can get like 16 resistance and 8 transmission (contact transmission) because a lot of transmission effect symptoms come with high resistance as well. And vice versa. (Specifically, mind restoration -2 resist, -3 trans, viral self adapt, +5 resist, inorganic biology, +4 resist, +3 transmission, sneezing, +3 resist, +4 transmission, fever, +3 resist, +2 transmission, coughing, +3 resist, +2 transmission.) Optimizing solely for resistance and mind restoration gives you 18 resistance and blood transmission.

@Redrover1760 I'm interested in this but I'm uncertain if there are any player-facing tooltips/info I'd need to update aside from just adding the code to make it work that way. You sound like an experienced viro player, anything I should be aware of?

**Comment by Time-Green** [comment]
I think it’s good to keep deep-rooted restricted to lobotomy surgeries, it’s weird to let the viro be a better brain surgeron than a brain surgeon

i’ll give this a day or two to give other people a chance to pop-in

**Comment by RaveRadbury** [comment]
I think that having multiple solutions to a problem is worthwhile, and I also think that viros aren't often put into positions where they need to make more than one virus in a round. 

In my experience of watching the viro main lifecycle, they perfect their virus and then they learn to make a virus without regen coma so miners won't get mad at them. Viros are generally disinterested in making multiple viruses in a round and I think that they might be better off if they had more reason to do so.

**Comment by Redrover1760** [comment]
> > As an alternative, you could add a new resistance tier to mind restoration, like 12 or 13, that allows it to cure deep rooted traumas instead of resistance 9 allowing it to cure deep rooted traumas, or something. Mind restoration gives -3 transmission and -2 resistance, so I think if one were to dedicate the virus to 13+ resistance and 7 transmission total for contact transmission it wouldn't be possible. At best it'd be fluid transmission..
> > Edit: I've done the math if one were to optimize solely resistance and transmission and mind restoration you can get like 16 resistance and 8 transmission (contact transmission) because a lot of transmission effect symptoms come with high resistance as well. And vice versa. (Specifically, mind restoration -2 resist, -3 trans, viral self adapt, +5 resist, inorganic biology, +4 resist, +3 transmission, sneezing, +3 resist, +4 transmission, fever, +3 resist, +2 transmission, coughing, +3 resist, +2 transmission.) Optimizing solely for resistance and mind restoration gives you 18 resistance and blood transmission.
> 
> @Redrover1760 I'm interested in this but I'm uncertain if there are any player-facing tooltips/info I'd need to update aside from just adding the code to make it work that way. You sound like an experienced viro player, anything I should be aware of?

Yeah. The only tooltip stuff I know exists would be the one in the pandemic machine. If a virus has its symptom, you can click on it to display what it does and its thresholds. Hovering over a threshold displays what that threshold does upon achieving it.

**Comment by Time-Green** [comment]
I edited the changelog to make it more clear what this means for obsessed

**Pull request merged**
